ripc-protocol-tcp/src/main/java/io/ripc/protocol/tcp/TcpInterceptor.java

```diff
@@ -0,0 +1,7 @@
+package io.ripc.protocol.tcp;
+
+public interface TcpInterceptor<I, O, II, OO> {
+
+    TcpHandler<II, OO> intercept(TcpHandler<I, O> handler);
```

Comment:
>> I was envisioning interceptors being analogous to Netty's pipeline feature.

Yes and No. Yes because it can change the types of input and output on the channel (read and write in netty pipeline) and no because it does not intercept any other channel lifecycle event apart from read/write.
Conceptually, interceptors are application level constructs where as netty channels are network level constructs thus needing more feedback around channel lifecycle.

>> How would we use this intercept mechanism to wrap each new connection dynamically

Have a look at the example [here](https://github.com/reactive-ipc/reactive-ipc-jvm/pull/19/files#diff-6327a044d2d50969539215f133a15744R19) which is wrapping over the next interceptor/handler in the chain and hence is in complete control of what connection to pass downstream.

>> especially those that come in as a result of an automatic reconnect after a client connection has died

I did not quiet understand, can you elaborate?